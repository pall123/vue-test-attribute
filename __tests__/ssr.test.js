import Vue from 'vue';
import {createRenderer} from 'vue-server-renderer';

import plugin from '../index';

Vue.config.productionTip = false;

describe('server side rendering', () => {
  describe('given a string value', () => {
    it('sets attribute to value', () => {
      const instance = new Vue(component('address'));
      const renderer = createRenderer({
        directives: {
          test: plugin.ssr()
        }
      });

      return renderer.renderToString(instance, (err, html) => {
        if (err) throw err;

        expect(html).toMatchSnapshot();
      });
    });
  });

  describe('given array value', () => {
    it('sets attribute to array contents separated by spaces', () => {
      const instance = new Vue(component(['address', 'address-1']));
      const renderer = createRenderer({
        directives: {
          test: plugin.ssr()
        }
      });

      return renderer.renderToString(instance, (err, html) => {
        if (err) throw err;

        expect(html).toMatchSnapshot();
      });
    });
  });

  describe('given false', () => {
    it('does not set attribute', () => {
      const instance = new Vue(component(false));
      const renderer = createRenderer({
        directives: {
          test: plugin.ssr()
        }
      });

      return renderer.renderToString(instance, (err, html) => {
        if (err) throw err;

        expect(html).toMatchSnapshot();
      });
    });
  });

  describe('settings.production === true', () => {
    it('does not add attribute to element', () => {
      const instance = new Vue(component('address'));
      const renderer = createRenderer({
        directives: {
          test: plugin.ssr({
            production: true
          })
        }
      });

      return renderer.renderToString(instance, (err, html) => {
        if (err) throw err;

        expect(html).toMatchSnapshot();
      });
    });
  });

  describe('settings.production === false', () => {
    it('adds attribute to element', () => {
      const instance = new Vue(component('address'));
      const renderer = createRenderer({
        directives: {
          test: plugin.ssr({
            production: false
          })
        }
      });

      return renderer.renderToString(instance, (err, html) => {
        if (err) throw err;

        expect(html).toMatchSnapshot();
      });
    });
  });

  describe('settings.production not set', () => {
    let oldNodeEnv = process.env.NODE_ENV;

    beforeEach(() => {
      process.env.NODE_ENV = 'production';
    });

    afterEach(() => {
      process.env.NODE_ENV = oldNodeEnv;
    });

    it('uses process.env.NODE_ENV to determine production', () => {
      const instance = new Vue(component('address'));
      const renderer = createRenderer({
        directives: {
          test: plugin.ssr({})
        }
      });

      return renderer.renderToString(instance, (err, html) => {
        if (err) throw err;

        expect(html).toMatchSnapshot();
      });
    });
  });
});

function component(name) {
  return {
    template: `<div v-test="name"></div>`,
    props: ['name'],
    propsData: {
      name
    }
  };
}