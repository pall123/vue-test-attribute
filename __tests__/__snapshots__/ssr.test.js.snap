// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server side rendering given a string value sets attribute to value 1`] = `<div data-test="address"></div>`;

exports[`server side rendering given array value sets attribute to array contents separated by spaces 1`] = `<div data-test="address address-1"></div>`;

exports[`server side rendering given false does not set attribute 1`] = `<div></div>`;

exports[`server side rendering settings.production === false adds attribute to element 1`] = `<div data-test="address"></div>`;

exports[`server side rendering settings.production === true does not add attribute to element 1`] = `<div></div>`;

exports[`server side rendering settings.production not set uses process.env.NODE_ENV to determine production 1`] = `<div></div>`;
